baseapp/baseapp.go

```diff
@@ -811,3 +802,54 @@ func createEvents(msg sdk.Msg) sdk.Events {
 
 	return sdk.Events{msgEvent}
 }
+
+func (app *BaseApp) prepareProposal(req abci.RequestPrepareProposal) ([][]byte, error) {
+	memTxs, selectErr := app.mempool.Select(req.Txs, req.MaxTxBytes)
+	if selectErr != nil {
+		panic(selectErr)
+	}
+	var txsBytes [][]byte
+	var txs []sdk.Tx
+	for _, memTx := range memTxs {
+		bz, encErr := app.txEncoder(memTx)
+		if encErr != nil {
+			panic(encErr)
+		}
+		txsBytes = append(txsBytes, bz)
+		txs = append(txs, memTx.(sdk.Tx))
+	}
+	ctx := app.checkState.ctx
+	err := app.checkTxsValidity(ctx, txs, txsBytes)
+	if err != nil {
+		return nil, err
+	}
+	return txsBytes, nil
+}
+
+func (app *BaseApp) processProposal(req abci.RequestProcessProposal) error {
+	ctx := app.checkState.ctx
```

Comment:
Question on this context, is `app.checkState.ctx` unaffected by any `CheckTx` executed _after_ the last block was delivered?